sql/src/main/java/io/crate/execution/ddl/TransportAnalyzeAction.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.execution.ddl;
+
+import io.crate.execution.support.MultiActionListener;
+import io.crate.execution.support.NodeAction;
+import io.crate.execution.support.NodeActionRequestHandler;
+import io.crate.execution.support.OneRowActionListener;
+import io.crate.planner.TableStatsService;
+import org.elasticsearch.action.ActionListenerResponseHandler;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.node.DiscoveryNodes;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.Singleton;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * TODO: This currently calls tableStatsService.updateStats on each node, which means each node by itself will
+ * generate the stats and populate it locally.
+ *
+ * A more efficient way of doing things would be to fetch the updated stats to the coordinator node and then publish
+ * the stats to all other nodes.
+ *
+ * Currently we do a sys.shards query to generate the stats, so this approach might be good enough for now?
+ */
```

Comment:
@kovrus @seut any inputs to this?

Reply:
I think I'll keep it as is in this draft. Once we extend the stats with more information we might want to change how it is generated as well anyway, so we can adapt this then. (E.g. maybe the TableStatsService will call analyze periodically, and analyze will do something else entirely)

Reply:
Yes I'm fine to keep this as is for now. As you mentioned already we may change this anyway later on.